app.js

```diff
@@ -134,14 +151,26 @@ router.post('/account/recoveryMethods', checkAccountOwnership, async ctx => {
     ctx.body = await recoveryMethodsFor(account);
 });
 
-async function withAccount(ctx, next) {
-    const { accountId } = ctx.request.body;
-    ctx.account = await models.Account.findOne({ where: { accountId } });
-    if (ctx.account) {
-        await next();
-        return;
+function getWithSequelizeAccountHandler(source) {
+    if (source !== 'body' && source !== 'params') {
+        throw new Error('invalid source for accountId provided to getWithSequelizeAccountHandler()');
     }
-    ctx.throw(404, `Could not find account with accountId: '${accountId}'`);
+
+    return async function withSequelizeAccount(ctx, next) {
+        let accountId;
```

Comment:
My first iteration was actually this with implicit fallback logic, but I created the middleware creator thunk for a few reasons:

1. I think that it is quicker to to reason about the flow of data at the route composition site when it's explicit what the middleware will use as its source of truth at the point where we create the middleware.
2. I think its safer and less likely to introduce subtle bugs as we add more routes if the person designing the route explicitly chooses which source to initialize the `ctx.sequelizeAccount` from, rather than having implicit fallback behavior that can result in the source of truth for the `sequelizeAccount` being different across individual requests to the same route depending on the content of the URL and the payload being sent.  In particular, I was thinking about any POST route with an optional `accountId` parameter, and how easy it would be to have the middleware target the wrong account for some requests in that case.  If we **never** have an optional `param` for `accountId` on any route then the the implicit fallback logic would be fine flexibility-wise, but I don't like relying on this kind of incidental behavior when there could well be a route like that in the future, which would require refactoring the implicit fallback to use this pattern anyway (I'd prefer to just touch this once middleware once).

If you have a strong opinion on this, I'll remove the interim middleware creator function entirely, but I think having the developer make _explicit_ choices about what data to use to initialize the sequelizeAccount is preferable, and the complexity cost here (like you mentioned, it's very common to have middleware creator thunks with arguments) is minimal.